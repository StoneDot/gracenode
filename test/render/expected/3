<!DOCTYPE html>
<html>
<html>
<head>
<meta charset="utf8" />
<script type="text/javascript">window.gracenode={"keys":["one","two","three"],"data":{"one":{"type":"string","value":"One"},"two":{"type":"number","value":2},"three":{"type":"array","value":[3]}}};(function(window){vardomain=window.location.origin;//exposewindow.request=request;/*options:{sendAsBinary:<boolean>,mimeType:<string>,headers:<object>}*/functionrequest(url,method,params,options,cb){url=domain+url;varreq;if(!params){params={};}if(!options){options={};}url=window.encodeURI(url);//sadbutwehavetodealwithIE<7if(!window.XMLHttpRequest){//sosomeoneisignorantenoughtobeuseIE...req=newwindow.ActiveXObject('Microsoft.XMLHTTP');}else{req=newwindow.XMLHttpRequest();}//setuprequestparameterstry{params=setupParams(params);}catch(exception){returncb(exception);}//setuprequestobjectreq.overrideMimeType(options.mimeType||'text');req.open(method,url,true);//defaultcontenttypeheaderreq.setRequestHeader('Content-Type','application/x-www-form-urlencoded');//ifweneedtosendmoreheadersif(options.headers){for(varnameinoptions.headers){req.setRequestHeader(name,options.headers[name]);}}//setupthelistenerreq.onreadystatechange=function(){handleResponse(req,cb);};//nowsendtherequestif(options.sendAsBinary){returnreq.sendAsBinary(params);}req.send(params);}functionsetupParams(params){varstr='';for(varkeyinparams){if(str!==''){str+='&';}str+=window.encodeURIComponent(key)+'='+prepareParamValue(params[key]);}returnstr;}functionprepareParamValue(param){if(typeofparam==='object'){returnwindow.encodeURIComponent(JSON.stringify(param));}returnwindow.encodeURIComponent(param);}functionhandleResponse(req,cb){if(req.readyState===4){//wenowhavetheresponsebackvarerror;varres=null;//evaluateresponsestatuscodeif(req.status>399){//errorerror=newError(req.status);}//dealwithresponsebodyif(req.responseText){//isitaJSON?try{res=JSON.parse(req.responseText);}catch(e){//itisnotaJSONres=req.responseText;}}cb(error,res,req.status);}}}(window));(function(){varTAGS={SRC:'data-gn-src',LOCAL:'data-gn-local'};varTAG_EXEC_MAP={};TAG_EXEC_MAP[TAGS.SRC]=getRemoteData;TAG_EXEC_MAP[TAGS.LOCAL]=getLocalData;document.EventListener('DOMContentLoaded',parse,false);window.render=function(target){parse(null,target);};functionparse(event,target){for(varkeyinTAGS){finder(TAGS[key],target);}}functionfinder(tagName,target){target=target||document;varlist=target.querySelectorAll('['+tagName+']')||[];varexec=TAG_EXEC_MAP[tagName];for(vari=0,len=list.length;i<len;i++){exec(list[i],tagName);}}functiongetRemoteData(elm,path){window.request(path,'GET',{},{},function(error,data){if(error){console.log(path,error);return;}vardom=buildDOM(data);elm.appendChild(dom);});}//example:gracenode.myDatafunctiongetLocalData(elm,str){varlist=str.split('.');vardata=window;for(vari=0,len=list.length;i<len;i++){data=data[list[i]];}if(!data||data===window){console.error(str,'InvalidLocalData');}vardom=buildDOM(data);elm.appendChild(dom);}functionbuildDOM(data){switch(typeofdata){case'object':if(Array.isArray(data)){returnbuildList(data);}returnbuildTwoColumnTable(data);default:returnbuildText(data);}}functionbuildList(data){varcontainer;vari;varlen;//alistofmap=tableif(typeofdata[0]==='object'){varrow;varlabels=[];container=document.createElement('table');container.className='list-table';//tablelabelsrow=document.createElement('tr');for(i=0,len=data.length;i<len;i++){for(varnameindata[i]){if(labels.indexOf(data[i][name])!==-1){continue;}varth=document.createElement('th');th.textContent=data[i][name];row.appendChild(th);}}container.appendChild(row);//tablerowsfor(i=0,len=data.length;i<len;i++){row=document.createElement('tr');vartd=document.createElement('td');content=buildDOM(data[i]);td.appendChild(td);row.appendChild(td);container.appendChild(row);}//donereturncontainer;}//alistoftext=listcontainer=document.createElement('ul');container.className='list-container';for(i=0,len=data.length;i<len;i++){varitem=document.createElement('li');item.appendChild(buildDOM(data[i]));container.appendChild(item);}//donereturncontainer;}functionbuildTwoColumnTable(data){varcontainer=document.createElement('table');container.className='two-col-table';for(varnameindata){varrow=document.createElement('tr');varlabel=document.createElement('th');label.textContent=name;row.appendChild(label);varvalue=document.createElement('td');value.appendChild(buildDOM(data[name]));row.appendChild(value);}returncontainer;}functionbuildText(data){varcontainer=document.createElement('div');container.className='text'+(typeofdata==='number'?'number':'');container.textContent=data;returncontainer;}}());</script>
</head>
<body>
<div>class&nbps;name:&nbsp;one</div><div>class&nbps;name:&nbsp;two</div><div>class&nbps;name:&nbsp;three</div>
<div>string:One</div><div>number:2</div><div>array:[3]</div>
</body>
</html>
</html>
