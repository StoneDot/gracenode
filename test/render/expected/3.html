<!DOCTYPE html>
<html>
<html>
<head>
<meta charset="utf8" />
<script type="text/javascript">window.gracenode={"keys":["one","two","three"],"data":{"one":{"type":"string","value":"One"},"two":{"type":"number","value":2},"three":{"type":"array","value":[3]}}};</script>
</head>
<body>
<div>class&nbps;name:&nbsp;one</div><div>class&nbps;name:&nbsp;two</div><div>class&nbps;name:&nbsp;three</div>
<div>string:One</div><div>number:2</div><div>array:[3]</div>
</body>
</html>
</html>
